tests/supervisors/test_signal.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import signal
+from asyncio import Event
+
+import httpx
+import pytest
+
+from tests.utils import run_server
+from uvicorn import Server
+from uvicorn.config import Config
+
+
+@pytest.mark.anyio
+async def test_sigint_finish_req(unused_tcp_port: int):
+    """
+    1. Request is sent
+    2. Sigint is sent to uvicorn
+    3. Shutdown sequence start
+    4. Request is finished before timeout_graceful_shutdown=1
+
+    Result: Request should go through, even though the server was cancelled.
+    """
+
+    server_event = Event()
+
+    async def wait_app(scope, receive, send):
+        await send({"type": "http.response.start", "status": 200, "headers": []})
+        await send({"type": "http.response.body", "body": b"start", "more_body": True})
+        await server_event.wait()
+        await send({"type": "http.response.body", "body": b"end", "more_body": False})
+
+    config = Config(
+        app=wait_app, reload=False, port=unused_tcp_port, timeout_graceful_shutdown=1
+    )
+    server: Server
+    async with run_server(config) as server:
+        async with httpx.AsyncClient() as client:
+            Event()
```

Comment:
Forgot to remove this event somehow, @Kludex. Sorry. Same for test below.

Reply:
np, can you open a PR removing it? 👀

Reply:
Of course!